indico/MaKaC/common/output.py

```diff
@@ -852,54 +852,48 @@ def confToXMLMarc21(self,conf,includeSession=1,includeContribution=1,includeMate
         version = "MARC21_ses-%s_cont-%s_mat-%s"%(includeSession,includeContribution,includeMaterial)
         obj = None
         if not overrideCache:
-            obj = self.cache.loadObject(version, conf)
+            obj = self.cache.loadObject(version, event.as_legacy)
         if obj:
             xml = obj.getContent()
         else:
             # No cache, build the XML
             temp = XMLGen(init=False)
-            self._confToXMLMarc21(conf,includeSession,includeContribution,includeMaterial, out=temp)
+            self._event_to_xml_marc_21(event, includeSession, includeContribution, includeMaterial, out=temp)
             xml = temp.getXml()
             # save XML in cache
-            self.cache.cacheObject(version, xml, conf)
+            self.cache.cacheObject(version, xml, event.as_legacy)
         out.writeXML(xml)
 
-    def _confToXMLMarc21(self,conf,includeSession=1,includeContribution=1,includeMaterial=1,out=None):
+    def _event_to_xml_marc_21(self, event, includeSession=1, includeContribution=1, includeMaterial=1, out=None):
         if not out:
             out = self._XMLGen
 
         out.openTag("datafield",[["tag","245"],["ind1"," "],["ind2"," "]])
-        out.writeTag("subfield",conf.getTitle(),[["code","a"]])
+        out.writeTag("subfield", event.title, [["code", "a"]])
         out.closeTag("datafield")
 
         out.writeTag("leader", "00000nmm  2200000uu 4500")
         out.openTag("datafield",[["tag","111"],["ind1"," "],["ind2"," "]])
-        out.writeTag("subfield",conf.title,[["code","a"]])
-
-        # XXX: If there is ROOM and not LOCATION....There will be information missed.
-        for l in conf.getLocationList():
-            loc = ""
-            if l.getName() != "":
-                loc = conf.getLocation().getName()
-            if l.getAddress() != "":
-                loc = loc +", "+conf.getLocation().getAddress()
-
-            if conf.getRoom():
-                roomName = self._getRoom(conf.getRoom(), l)
-                loc = loc + ", " + roomName
-
-            if l.getName() != "":
-                out.writeTag("subfield",loc,[["code","c"]])
-
-        sd = conf.getStartDate()
-        ed = conf.getEndDate()
+        out.writeTag("subfield", event.title, [["code", "a"]])
+        event_location_info = []
+        if event.venue_name:
+            event_location_info.append(event.venue_name)
+        if event.address:
+            event_location_info.append(event.address)
+        event_room = event.room.name if event.room else event.room_name
```

Comment:
Not exactly. The `room_name` property will return the `room.full_name` in case `room` exists. We need `room.name` in this case.